gadgets/bpfstats/README.mdx

```diff
@@ -0,0 +1,205 @@
+`--enable-bpfstats` flag:
+
+First, we need to run an application that generates some events.
+
+<Tabs groupId="env">
+  <TabItem value="kubectl-gadget" label="kubectl gadget">
+```bash
+$ kubectl gadget deploy --enable-bpfstats
+```
+  </TabItem>
+
+  <TabItem value="ig-daemon" label="ig-daemon">
+```bash
+$ sudo ig daemon --enable-bpfstats
+```
+  </TabItem>
+</Tabs>
+
+Then, let's run the gadget:
+
+<Tabs groupId="env">
+  <TabItem value="kubectl-gadget" label="kubectl gadget">
+
+```bash
+$ kubectl gadget run bpfstats:%IG_TAG%
+NODENAME           GADGETID  GADGETNAME         GADGETIMAGE        PROGID PROGNAME               RUNTIME     RUNCOUNT    MAPMEMORY MAPC…
+```
+  </TabItem>
+
+  <TabItem value="ig-daemon" label="ig-daemon">
+```bash
+$ sudo gadgetctl run bpfstats:%IG_TAG%
+GADGETID    GADGETNAME           GADGETIMAGE          PROGID PROGNAME                     RUNTIME        RUNCOUNT       MAPMEMORY MAPCO…
+```
+  </TabItem>
+</Tabs>
+
+It won't print anything as there are not gadgets running on the system. Let's
+create a couple of gadgets:
+
+<Tabs groupId="env">
+  <TabItem value="kubectl-gadget" label="kubectl gadget">
+```bash
+$ kubectl gadget run --name mytraceopen trace_open:%IG_TAG% --detach
+$ kubectl gadget run --name mytraceexec trace_exec:%IG_TAG% --detach
+```
+  </TabItem>
+
+  <TabItem value="ig-daemon" label="ig-daemon">
+```bash
+$ sudo gadgetctl run --name mytraceopen trace_open:%IG_TAG% --detach
+$ sudo gadgetctl run --name mytraceexec trace_exec:%IG_TAG% --detach
+```
+  </TabItem>
+</Tabs>
+
+The bpfstats will print the statistics for these two gadgets:
+
+<Tabs groupId="env">
+  <TabItem value="kubectl-gadget" label="kubectl gadget">
+```bash
+$ kubectl gadget run bpfstats:%IG_TAG%
+NODENAME           GADGETID  GADGETNAME         GADGETIMAGE        PROGID PROGNAME               RUNTIME     RUNCOUNT    MAPMEMORY MAPC…
+minikube           6bcfcca7a mytraceopen        trace_open:%IG_TAG%     0                        2537846        10798      1710240     6
+minikube           040e3fc1d mytraceexec        trace_exec:%IG_TAG%     0                           8265           12      8742176     7
+```
+  </TabItem>
+
+  <TabItem value="ig-daemon" label="ig-daemon">
+```bash
+$ sudo gadgetctl run bpfstats:%IG_TAG%
+GADGETID    GADGETNAME           GADGETIMAGE          PROGID PROGNAME                     RUNTIME        RUNCOUNT       MAPMEMORY MAPCO…
+a624bdbf27f mytraceexec          trace_exec:%IG_TAG%       0                                    0               0         8742176      7
+bb95fa2c8ab mytraceopen          trace_open:%IG_TAG%       0                                95233             242         1710240
+```
+  </TabItem>
+</Tabs>
+
+By default the gadget prints consolidated information for the Gadgets. It's
+possible to get per-program statistics by using the `--programs` flag. In this
+mode, the statistics for eBPF programs not handled by Inspektor Gadget are shown
+as well.
```

Comment:
I'm wondering if we should use two flags for the two functionalities this flag provides:

- `per-program` to not summarize stats per gadget but per program
- `all` to show also programs that don't belong to any gadget

WDYT?

Reply:
I'm not totally convinced. I think it's too much. Perhaps having only `--all` will be enough (it prints all programs on the system, and enriches with gadget information the ones from Inspektor Gadget).

Reply:
I'm ok with just `all`.